Library/Homebrew/cmd/list.rb

```diff
@@ -86,7 +86,13 @@ def list
         ls_args << "-r" if args.r?
         ls_args << "-t" if args.t?
 
-        safe_system "ls", *ls_args, HOMEBREW_CELLAR
+        if !$stdout.tty? && !args.formula?
+          odeprecated "`brew list` to only list formulae", "`brew list --formula`"
```

Comment:
this `odeprecated` call was converted to `odisabled` in #9209; I've opened https://github.com/Homebrew/brew/pull/9357 to revert that change so it will remain `odeprecated` for 2.6.0